materialize/resources/resource_materialized_view.go

```diff
@@ -0,0 +1,241 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmoiron/sqlx"
+)
+
+var materializedViewSchema = map[string]*schema.Schema{
+	"name": {
+		Description: "The identifier for the materialized view.",
+		Type:        schema.TypeString,
+		Required:    true,
+	},
+	"schema_name": {
+		Description: "The identifier for the materialized view schema.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "public",
+		ForceNew:    true,
+	},
+	"database_name": {
+		Description: "The identifier for the materialized view database.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "materialize",
+		ForceNew:    true,
+	},
+	"qualified_name": {
+		Description: "The fully qualified name of the materialized view.",
+		Type:        schema.TypeString,
+		Computed:    true,
+	},
+	"in_cluster": {
+		Description: "The cluster to maintain this materialized view.",
+		Type:        schema.TypeString,
+		Optional:    true,
+	},
+	"select_stmt": {
+		Description: "The SQL statement to create the materialized view.",
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+	},
+}
+
+func MaterializedView() *schema.Resource {
+	return &schema.Resource{
+		Description: "A materialized view persists in durable storage and can be written to, updated and seamlessly joined with other views, views or sources.",
+
+		CreateContext: materializedViewCreate,
+		ReadContext:   materializedViewRead,
+		UpdateContext: materializedViewUpdate,
+		DeleteContext: materializedViewDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: materializedViewSchema,
+	}
+}
+
+type MaterializedViewBuilder struct {
+	materializedViewName string
+	schemaName           string
+	databaseName         string
+	inCluster            string
+	selectStmt           string
+}
+
+func (b *MaterializedViewBuilder) qualifiedName() string {
+	return QualifiedName(b.databaseName, b.schemaName, b.materializedViewName)
+}
+
+func newMaterializedViewBuilder(materializedViewName, schemaName, databaseName string) *MaterializedViewBuilder {
+	return &MaterializedViewBuilder{
+		materializedViewName: materializedViewName,
+		schemaName:           schemaName,
+		databaseName:         databaseName,
+	}
+}
+
+func (b *MaterializedViewBuilder) InCluster(inCluster string) *MaterializedViewBuilder {
+	b.inCluster = inCluster
+	return b
+}
+
+func (b *MaterializedViewBuilder) SelectStmt(selectStmt string) *MaterializedViewBuilder {
+	b.selectStmt = selectStmt
+	return b
+}
+
+func (b *MaterializedViewBuilder) Create() string {
+	q := strings.Builder{}
+	q.WriteString(`CREATE`)
```

Comment:
Nit: But could be done in fewer lines, something like:
```
	q.WriteString(fmt.Sprintf(`CREATE MATERIALIZED VIEW %s`, b.qualifiedName()))

	if b.inCluster != "" {
		q.WriteString(fmt.Sprintf(` IN CLUSTER %s`, b.inCluster))
	}

	q.WriteString(fmt.Sprintf(` AS %s;`, b.selectStmt))
	return q.String()
``

Reply:
neat! thank you for this suggestion!